keystore/src/connection/platform/wasm/storage.rs

```diff
@@ -26,9 +26,83 @@ use crate::{
 
 use super::WasmConnection;
 
+type InMemoryDB = HashMap<String, HashMap<Vec<u8>, JsValue>>;
+
 pub enum WasmStorageWrapper {
     Persistent(idb::Database),
-    InMemory(HashMap<String, HashMap<Vec<u8>, JsValue>>),
+    InMemory(Rc<RefCell<InMemoryDB>>),
+}
+
+// The lifetime is to comply with the sqlite implementation.
+pub enum WasmStorageTransaction<'a> {
+    Persistent {
+        tx: idb::Transaction,
+        cipher: &'a aes_gcm::Aes256Gcm,
+    },
+    InMemory {
+        db: Rc<RefCell<InMemoryDB>>,
+        cipher: &'a aes_gcm::Aes256Gcm,
+    },
+}
+
+impl<'a> WasmStorageTransaction<'a> {
+    pub(crate) async fn commit_tx(self) -> CryptoKeystoreResult<()> {
+        let Self::Persistent {
+            tx: transaction,
+            cipher: _cipher,
+        } = self
+        else {
+            return Ok(());
+        };
+        transaction.commit()?.await?;
+        Ok(())
+    }
+
+    pub(crate) async fn delete(&self, collection_name: &'static str, id: impl AsRef<[u8]>) -> CryptoKeystoreResult<()> {
+        match self {
+            WasmStorageTransaction::Persistent { tx, cipher: _cipher } => {
+                let store = tx.object_store(collection_name)?;
+                let k = Uint8Array::from(id.as_ref());
+                store.delete(JsValue::from(k))?.await?;
+            }
+            WasmStorageTransaction::InMemory { db, cipher: _cipher } => {
+                db.borrow_mut().entry(collection_name.into()).and_modify(|store| {
+                    let result = store.remove(id.as_ref());
+                    debug_assert!(result.is_some());
+                });
+            }
+        }
+        Ok(())
+    }
+
+    pub(crate) async fn save<R: Entity<ConnectionType = WasmConnection> + 'static>(
+        &self,
+        collection_name: &'static str,
```

Comment:
This argument is unecessary, you can just use `R::COLLECTION_NAME` below

Reply:
Fixed in: https://github.com/wireapp/core-crypto/pull/695/commits/af6abef688cd89ab886d6aee49cb1f089c7dc3ba